pisa/analysis/analysis.py

```diff
@@ -266,17 +266,22 @@ def get_separate_t23_octant_params(hypo_maker,inflection_point) :
 
     # Get the octant definition
     octants = (
-        (theta23.range[0],inflection_point) ,
-        (inflection_point,theta23.range[1])
+        (theta23.range[0], inflection_point) ,
+        (inflection_point, theta23.range[1])
         )
 
-    # If theta23 is very close to maximal (e.g. the transition between octants)
-    # offset it slightly to be clearly in one octant (note that fit can still
-    # move the value back to maximal)
-    tolerance = 1. * ureg.degree
-    if np.isclose(theta23.value.m_as("degree"),45.,atol=tolerance.m_as("degree")) : 
-        theta23.value -= tolerance
-
+    # If theta23 is maximal (e.g. the transition between octants) or very close 
+    # to it, offset it slightly to be clearly in one octant (note that fit can 
+    # still move the value back to maximal). The reason for this is that 
+    # otherwise checks on the parameter bounds (which include a margin for 
+    # minimizer tolerance) can an throw exception.
+    tolerance = 0.1 * ureg.degree
```

Comment:
Why wouldn't a value within `tolerance` of maximal simply be placed at `tolerance` away from maximal, rather than having `tolerance` added (subtracted) from the value?

E.g., if a user (or whatever sets it) asks for t23=44.901 deg, doesn't setting t23=44.9 deg seem like the better "approximation" to what the user wants than 44.801 deg?

I don't perceive any benefit to having a "linear" shifting behavior for this operation. But if I'm missing something, please let me know.

Reply:
What do you think of the solution in my PR? https://github.com/IceCubeOpenSource/pisa/pull/587/files#diff-5f6e672ea43f8f20cf02b7c62bafe16fR151 (and following)

Reply:
Hi Justin, I think what your suggesting the code should be doing is what it is doing. E.g. If you are closer than `tolerance` to 45 degrees, you are moved to 45 +/- tolerance (where the sign is chosen to stay in the same octant).

Reply:
It looks to me like Thomas' solution is doing the same thing. So either implementation is fine by me.

Reply:
You're right, Tom; I misread what the code was doing. So we're good on this.